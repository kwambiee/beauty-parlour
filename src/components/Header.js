import React from 'react';

const Header = () => (
  <div className="header">
    <h1>
      DROP-DEAD
      {' '}
      <br />
      GORGEOUS
    </h1>
    <button type="button" className="button">
      SHOP NOW
    </button>
  </div>
);
export default Header;
