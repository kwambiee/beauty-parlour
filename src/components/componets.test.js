import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import Brand from './brand';
import Product from './product';
import Header from './Header';
import Store from '../Redux/ConfigureStore';

export const brandProvider = () => (
  <Provider store={Store}>
    <Brand />
  </Provider>
);

export const headerProvider = () => (
  <Provider store={Store}>
    <Header />
  </Provider>
);

export const productProvider = () => (
  <Provider store={Store}>
    <Product />
  </Provider>
);

describe('Check Brand Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<brandProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
describe('Check header Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<headerProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
describe('Check product Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<productProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

test('Link matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <Link to="/" />
    </Router>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
