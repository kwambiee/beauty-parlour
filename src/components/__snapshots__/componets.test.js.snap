// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Brand Component renders correctly 1`] = `<brandProvider />`;

exports[`Check Brand Component renders correctly 2`] = `<headerProvider />`;

exports[`Check Brand Component renders correctly 3`] = `<productProvider />`;

exports[`Check header Component renders correctly 1`] = `<headerProvider />`;

exports[`Check product Component renders correctly 1`] = `<productProvider />`;

exports[`Link matches snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
/>
`;
